File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.go
func judgeSquareSum(c int) bool {
    a := 0;
    b := int(math.Sqrt(float64(c)));

    for a <= b {
        sum := int64(a)*int64(a)+int64(b)*int64(b);
        if sum == int64(c) {
            return true;
        } else if sum > int64(c) {
            b--;
        } else {
            a++;
        }
    }
    return false;
}

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.cpp
class Solution {
public:
    bool judgeSquareSum(int c) {
        long long a = 0;
        long long b = sqrt(c);

        while (a <= b) {
            long long sum = a*a+b*b;
            if (sum == c) {
                return true;
            } else if (sum > c) {
                b--;
            } else {
                a++;
            }
        }
        return false;
    }
};